code/modules/client/preference_setup/general/04_equipment.dm

```diff
@@ -48,70 +48,83 @@
 
 /datum/category_item/player_setup_item/general/equipment/gather_save_parameters()
 	return list(
-		"underwear" = pref.underwear,
-		"undershirt" = pref.undershirt,
-		"socks" = pref.socks,
+		"all_underwear" = pref.all_underwear,
+		"all_underwear_metadata" = pref.all_underwear_metadata,

```

Comment:
You'll have to use JSON or params2list to store a DM list. Straight uploading a list does not work.

`"all_underwear_metadata" = json_encode(pref.all_underwear_metadata),`